osu.Desktop/DiscordRichPresenceClient.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using DiscordRPC;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Logging;
+using osu.Game.Online.API;
+using osu.Game.Rulesets;
+using osu.Game.Users;
+using User = osu.Game.Users.User;
+
+namespace osu.Desktop
+{
+    internal class DiscordRichPresenceClient : Component
+    {
+        private const string client_id = "563024054391537674";
+
+        private Bindable<User> user;
+
+        private readonly DiscordRpcClient client = new DiscordRpcClient(client_id);
```

Comment:
`DiscordRpcClient` is `IDisposable`
see https://github.com/Lachee/discord-rpc-csharp/blob/master/DiscordRPC/DiscordRpcClient.cs

Hence I guess `DiscordRichPresenceClient` should implement `IDisposable` too.